src/main/java/org/jabref/gui/shared/SharedDatabaseLoginDialogView.java

```diff
@@ -0,0 +1,320 @@
+                                                                                           Localization.lang("Cancel"));
+                if (!overwriteFilePressed) {
+                    tbFolder.requestFocus();
+                    return;
+                }
+            }
+        }
+
+        setLoadingConnectButtonText(true);
+
+        try {
+            SharedDatabaseUIManager manager = new SharedDatabaseUIManager(frame);
+            BasePanel panel = manager.openNewSharedDatabaseTab(connectionProperties);
+            setPreferences();
+
+            if (!tbFolder.getText().isEmpty()) {
+                try {
+                    new SaveDatabaseAction(panel, Paths.get(tbFolder.getText())).runCommand();
+                } catch (Throwable e) {
+                    LOGGER.error("Error while saving the database", e);
+                }
+            }
+
+            return; // setLoadingConnectButtonText(false) should not be reached regularly.
+        } catch (SQLException | InvalidDBMSConnectionPropertiesException exception) {
+
+            frame.getDialogService().showErrorDialogAndWait(Localization.lang("Connection error"), exception);
+
+        } catch (DatabaseNotSupportedException exception) {
+            ButtonType openHelp = new ButtonType("Open Help", ButtonData.OTHER);
+
+            Optional<ButtonType> result = dialogService.showCustomButtonDialogAndWait(AlertType.INFORMATION,
+                                                                                      Localization.lang("Migration help information"),
+                                                                                      Localization.lang("Entered database has obsolete structure and is no longer supported.")
+                                                                                                                                       + "\n" +
+                                                                                                                                       Localization.lang("Click help to learn about the migration of pre-3.6 databases.")
+                                                                                                                                       + "\n" +
+                                                                                                                                       Localization.lang("However, a new database was created alongside the pre-3.6 one."),
+                                                                                      ButtonType.OK, openHelp);
+
+            result.filter(btn -> btn.equals(openHelp)).ifPresent(btn -> HelpAction.openHelpPage(HelpFile.SQL_DATABASE_MIGRATION));
+            result.filter(btn -> btn == ButtonType.OK).ifPresent(btn -> openSharedDatabase());
+
+        }
+
+        setLoadingConnectButtonText(false);
+    }
+
+    private void setLoadingConnectButtonText(boolean isLoading) {
+        btnConnect.setDisable(isLoading);
```

Comment:
The `viewModel` should have a boolean property `connectionInProgress` so that the functionality of this method can be realized by binding.